frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -33,7 +33,9 @@ import {
   Resources,
 } from './import-types';
 import { createPipelineForImportFlow } from './pipeline/pipeline-template-utils';
+import { getRandomChars } from '../pipelines/pipeline-resource/pipelineResource-utils';
 
+let generatedImageStreamName: string;
```

Comment:
Oh yikes... the reason this is outside of the function is that you're sharing it between functions... this is not ideal in anyway. Bump this variable up a scope and then pass it through to the functions as needed.

Looks like the goal of this is to revalue `name`... perhaps do it seamlessly at the top level by reworking the "formData" to include the repurposed name before it's passed down. This feels like a hack where one (I don't think) is needed.

Reply:
yeah, this was bad. have updated it. Thanks